### DISCUSSION TITLE
Mypy complains about QuerySet. Wrong  number of type arguments

### DISCUSSION THREAD

**Discussion by baojd42** [typeddjango/django-stubs#2279] [Q&A] [answered]
I use `QuerySet[MyModel]` to annotate the return type of a function, but mypy gives an errror "'QuerySet' expects 2 type arguments, but 1 given". I wonder why it requires 2 type arguments. I see a lot of usage with just one model type as argument to Queryset.

**Comment by intgr** [comment, marked as answer]
I think you're using an old unsupported version of mypy, which doesn't support TypeVar defaults. Current django-stubs requires mypy 1.11.x.

**Reply by baojd42** [reply]
Yes, you are right.  I forgot to mention that I am using tvscode mypy extension, which ships with a mypy ==1.6.0. I've then tested with a higher version of mypy in my projects's venv, the type error for QuerySet doesn't show up. I will set the extension to use mypy from the environment. Thanks a lot!

**Reply by Rastaalexander** [reply]
Jaja
